app/main.py

```diff
@@ -1,3 +1,56 @@
+from __future__ import annotations
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union[Distance, float, int]) -> Distance:
+        km = (
+            self.km + other
+            if isinstance(other, (float, int))
+            else self.km + other.km
+        )
+        return Distance(km)
+
+    def __iadd__(self, other: Union[Distance, float, int]) -> Distance:
+        self.km += other if isinstance(other, (float, int)) else other.km
+        return self
+
+    def __mul__(self, other: float | int) -> Distance:
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: float | int) -> Distance:
+        return Distance(round(self.km / other, 2))
+
+    def __lt__(self, other: Union[Distance, float, int]) -> bool:
+        return self.km < (
+            other if isinstance(other, (float, int)) else other.km
+        )
+
+    def __gt__(self, other: Union[Distance, float, int]) -> bool:
+        return self.km > (
+            other if isinstance(other, (float, int)) else other.km
+        )
+
+    def __eq__(self, other: Union[Distance, float, int]) -> bool:
+        return self.km == (
+            other if isinstance(other, (float, int)) else other.km
+        )
+
+    def __le__(self, other: Union[Distance, float, int]) -> bool:
+        return self.km <= (
```

Comment:
Ok so think about it: you have `less then or equal` method here. You already defined handling of `other` object for comparison in your `__lt__` method. Now thinking logically, if a < b it also means that b >= a. Now try to use this logic  here and in `__ge__`, i promise you will like it